atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/OpenTransaction.java

```diff
@@ -16,27 +16,13 @@
 
 package com.palantir.atlasdb.transaction.api;
 
-import com.palantir.atlasdb.metrics.Timed;
-
-public interface OpenTransaction extends Transaction {
-
-    /**
-     * Runs a provided task, commits the transaction, and performs cleanup. If no further work needs to be done with the
-     * transaction, a no-op task can be passed in.
-     *
-     * @return value returned by the task
-     */
-    @Timed
-    <T, E extends Exception> T finish(TransactionTask<T, E> task) throws E, TransactionFailedRetriableException;
+import java.io.Closeable;
 
+public interface OpenTransaction extends Transaction, Closeable {
     /**
-     * Like {@link #finish(TransactionTask)}, except runs a callback after the task has finished. This callback will
-     * not run while the transaction remains in an uncommitted state, but may run afterwards, regardless of whether
-     * the commit was successful, finished, or aborted.
-     *
-     * If the callback is run, it will run before any {@link Transaction#onSuccess(Runnable)} callbacks.
+     * Aborts the transaction if uncommitted and cleanups transaction state.
+     * All open transactions <b>must be closed</b>.
```

Comment:
Happy to add more details. The prior API did not clarify consequences for not calling `close()`.

Reply:
You're right the previous API didn't clarify consequences. I think we can just say not closing transactions after they are no longer in use may lead to arbitrary delays elsewhere in the system 😓